src/server/pps/server/pipeline_controller.go

```diff
@@ -50,12 +61,28 @@ func max(is ...int) int {
 // pipelineOp contains all of the relevent current state for a pipeline. It's
 // used by step() to take any necessary actions
 type pipelineOp struct {
-	m *ppsMaster
 	// a pachyderm client wrapping this operation's context (child of the PPS
 	// master's context, and cancelled at the end of step())
 	ctx          context.Context
+	pipeline     string
 	pipelineInfo *pps.PipelineInfo
 	rc           *v1.ReplicationController
+	namespace    string
+	env          serviceenv.ServiceEnv
+	etcdPrefix   string
+	pipelines    collection.PostgresCollection
+	// writeBump and deleteBump represent whether a write or delete operation should
+	// be executed once the active pipelineOp goroutine is complete.
+	writeBump  bool
+	deleteBump bool
```

Comment:
the pipelineOp methods maintain the invariant that at most one of `writeBump` and `deleteBump` may be true.

This represents the LAST operation recorded on the pipeline